### PR TITLE
ISSUE-6072 Add test for gl convolution and update shader so that it passes

### LINKED ISSUE #6072: Convolution filter index edge conditions are off by one
## Version
3.6.0

## Test Case
Unit tests as well as a fix added here: https://github.com/rfinck/fabric.js/blob/fix-convolution-indexing/test/unit/image_filters.js#L703 
I can make a pull request but there are some workarounds I had to do to get the Node tests to run that may or may not be desirable in the project.

## Information about environment
Nodejs or browser? Both 
Which browsers? Any

## Steps to reproduce
See unit test linked above, which applies the following convolution filter to the following data:
```
var filter = new fabric.Image.filters.Convolute({
  matrix: [
    0, 1, 0,
    1, 2, 1,
    0, 1, 0
  ]
});
var pixelData = new Uint8Array([
  0, 0, 0, 255, 0, 0, 0, 255, 0, 0, 0, 255,
  0, 0, 0, 255, 1, 2, 3, 255, 0, 0, 0, 255,
  0, 0, 0, 255, 0, 0, 0, 255, 0, 0, 0, 255,
]);
```
## Expected Behavior
```
var expected = new Uint8Array([
  0, 0, 0, 255, 1, 2, 3, 255, 0, 0, 0, 255,
  1, 2, 3, 255, 2, 4, 6, 255, 1, 2, 3, 255,
  0, 0, 0, 255, 1, 2, 3, 255, 0, 0, 0, 255,
]);
```
## Actual Behavior
```
var actual = new Uint8Array([
  0, 0, 0, 255, 1, 2, 3, 255, 0, 0, 0, 255,
  1, 2, 3, 255, 2, 4, 6, 255, 0, 0, 0, 255,
  0, 0, 0, 255, 0, 0, 0, 255, 0, 0, 0, 255,
]);
```

### PR DESCRIPTION
**Pull request by rfinck**
This is the gl half of #6072 ; the js part is in #6088 since it's a more straightforward change to the tests. It's not clear how the changes here will perform in ci so partly creating this PR to check.

Here I added a test for the gl convolution filter, which required some changes to the shaders to pass in Node. The issue the previous shaders had may not have been a problem when run in the browser.

close #6072

### CODE REVIEW

**Comment by rfinck** [comment]
It looks like tests are failing on Node.js 12 and up due to a known issue: https://github.com/stackgl/headless-gl#supported-platforms-and-nodejs-versions

Locally, tests pass for my with Node.js 8 and Node.js 10, as well as with Chrome.
I can reproduce the failures with Firefox and Node.js > 12.

File: src/filters/convolute_filter.class.js [hunk 1]
```diff
@@ -311,7 +311,7 @@
      */
     getUniformLocations: function(gl, program) {
       return {
-        uMatrix: gl.getUniformLocation(program, 'uMatrix'),
+        uMatrix: gl.getUniformLocation(program, 'uMatrix[0]'),
```
Comment (asturur) [inline src/filters/convolute_filter.class.js]: can you explain me this? because i have no idea what does it do.
I would expect this to be just a reference to how is named the variable in the shader code.

**Comment by asturur** [comment]
i have to look a bit at how gl libray work and if it can be integrated with node-canvas or jsdom

File: src/filters/convolute_filter.class.js (hunk 1 above)
Comment (rfinck) [reply src/filters/convolute_filter.class.js]: Yes it is a reference to how it's named, my understanding is that either referencing it as `'uMatrix'` or as its first element `'uMatrix[0]'` should work but only the second one worked when I was gl in tests; both seem to work when using it normally.

**Comment by rfinck** [comment]
By running Xvfb in travis, unit tests now pass for Node < 13. The remaining issues are Firefox and Node 13 where the gl install errors.

Job | Node.js | ENV | State
-- | -- | -- | --
6643.3 | 12 | LAUNCHER=Chrome | passed
6643.4 | 12 | LAUNCHER=Firefox | failed
6643.5 | 13 | LAUNCHER=Node | errored
6643.6 | 12 | LAUNCHER=Node | passed
6643.7 | 10 | LAUNCHER=Node | passed
6643.8 | 8 |   | passed

**Comment by asturur** [comment]
When running in firefox we should not need the GL module at all, its requiring should be conditional.
That is why i wanted to know if we can just connect it to JSDOM and let JSDOM emulate a normal canvas with webgl support.
